lib/task_listener.js

```diff
@@ -110,21 +108,23 @@ export default class TaskListener extends EventEmitter {
     return cancelListener;
   }
 
+  get availableCapacity() {
+    return Math.max(this.capacity - this.runningTasks.length, 0);
```

Comment:
The only time that I can think of this happening (unless something goes greatly wrong) is when capacity is set to 0 when a node is being terminated and anything relying on tasklistener.avaiableCapacity checks it before the running tasks are aborted.